### PR TITLE
Required form

### LINKED ISSUE #62: Required Form :Functionality
As a user when I arrive to the page I want to be able to create a user by email, birthday and name

### PR DESCRIPTION
**Pull request by taylorjohnson141**
### Code Highlights:
Adds the required form functionality to the application

### Where should the reviewer start?
 Required form.tsx
 
### Have Tests Been Added?
- [x] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?
This will cause the createUser mutation to be fired which will create a user and generate a userId to createEras and events.

### Issues:
* Closes #62 

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: src/Components/FormStartPage/FormStartPage.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Link } from "react-router-dom";
+type FormStartPageProps = {
+  userId: number;
+  userName: string;
+};
+let FormStartPage = (props: FormStartPageProps) => {
```
Comment (aemcdonald) [inline src/Components/FormStartPage/FormStartPage.tsx:7]: Will you remove this console log before we merge to main please?

File: src/Components/FormStartPage/FormStartPage.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  console.log(props.userName);
+  return (
+    <section>
+      <p>
+        As you know life is filled with choices. We've decided to give you
+        another. You can either start off editend your calender from a blank
```
Comment (aemcdonald) [inline src/Components/FormStartPage/FormStartPage.tsx]: Looks like a small typo here :)

File: src/Components/FormStartPage/FormStartPage.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        calender or you can start off editing on a calender with some events. If
+        you choose the prefilled calender
+      </p>
+      <Link to="/calender">
+        <button>Start Calender</button>
```
Comment (aemcdonald) [inline src/Components/FormStartPage/FormStartPage.tsx:17]: One more typo: Calender should be Calendar with an A

File: src/Components/RequiredForm/RequiredForm.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  }
+  `
+let [createUser, { data,loading,error,called }] = useMutation(create_user);
+
+let sendUser = async () =>{
+  
+  let Bday = answers[1]
+  
+  let finalBday = Bday.split('-').reverse().join('-')
```
Comment (aemcdonald) [inline src/Components/RequiredForm/RequiredForm.tsx:29]: Do you think we should talk to backend about allowing any date format or keep this as is? Either way, nice error handling here for making sure the date is formatted correctly!

File: src/Components/TextQuestion/TextQuestion.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  const [text, updateText] = useState('')
+  const [answer, saveAnswer] = useState('');
+
+  const handleChange =  (event: React.ChangeEvent<HTMLInputElement>) => {
+    updateText(event.target.value)
+    saveAnswer(event.target.value)
+
+    
+  };
+  let type = 'text'
```
Comment (aemcdonald) [inline src/Components/TextQuestion/TextQuestion.tsx:24]: Awesome job conditionally rendering the correct input here

**Review by aemcdonald — CHANGES_REQUESTED** [review]
I noticed a couple typos to fix before merging. Other than that it looks great! Awesome job getting the required questions up & running. Super exciting that you got the mutation working as well!

**Review by aemcdonald — APPROVED** [review]
Looks great! Thank you for making those changes!

File: src/Components/FormStartPage/FormStartPage.tsx (hunk 2 above)
Comment (krogowsk531) [reply src/Components/FormStartPage/FormStartPage.tsx]: Love this!

**Review by krogowsk531 — APPROVED** [review]
This was really difficult!  Thanks for getting it done!

**Pull request merged**